sherpa/astro/ui/utils.py

```diff
@@ -1458,19 +1458,79 @@ def calc_staterror(data):
                 datas.elo, datas.ehi = calc_err(datas)
             datas.staterror = calc_staterror(datas)
 
-    # DOC-NOTE: also in sherpa.utils
+    def _load_data(self, id, datasets):
+        """Load one or more datasets.
+
+        Used by load_data and load_pha.
+
+        Parameters
+        ----------
+        id : int or str, optional
+           The identifier for the data set to use. For multi-dataset
+           files, currently only PHA2, the id value indicates the
+           first dataset: if it is an integer then the numbering
+           starts at id, and if a string then a suffix of 1 to n is
+           added.  If not given then the default identifier is used,
+           as returned by `get_default_id`.
+        datasets : Data instance or iterable of Data instances
+           The data to load, either as a single item or, for
+           multiple-dataset files, an iterable of them.
+
+        """
+
+        if not numpy.iterable(datasets):
+            self.set_data(id, datasets)
+            return
+
+        # One issue with the following is that if there's
+        # only one dataset in phasets and id is a string then the
+        # output will be "foo1" rather than "foo" (when
+        # id="foo").  DJB thinks we can live with this.
+        #
```

Comment:
Will this happen even for PHA type I datasets? If so, I think we should fix it. People would be a little surprised if every dataset named with a string suddenly has a "1" at the end. If (as a I understand that) this happens only for PHAII datasets, then we should declare that a feature and move this comment into the docstring.

Reply:
This only happens when unpacking a dataset returns multilpe data objects, and this only happens for a PHA2 dataset. It would have been obvious in our tests, but I did add an explicit check for this which is already in the master branch: https://github.com/sherpa/sherpa/blob/40f27543a380aec34e2a5cab264c47a3bbd58a8e/sherpa/astro/ui/tests/test_astro_ui.py#L253

Unfortunately, for reasons which 'll mention on the main discussion, we don't have a test of the 'load a single file from a PHA2' case.

I had thought the documentation was already clear about the id naming scheme, but I am way too close to it!

Reply:
Also note: this case may never happen (a single file being handled by the multi-file code).

Reply:
Then we won't worry about it.